pkg/statistics/handle/syncload/stats_syncload.go

```diff
@@ -223,46 +226,59 @@ func (s *statsSyncLoad) HandleOneTask(sctx sessionctx.Context, lastTask *statsty
 	} else {
 		task = lastTask
 	}
+	result := stmtctx.StatsLoadResult{Item: task.Item.TableItemID}
 	resultChan := s.StatsLoad.Singleflight.DoChan(task.Item.Key(), func() (any, error) {
-		return s.handleOneItemTask(sctx, task)
+		err := s.handleOneItemTask(sctx, task)
+		return nil, err
 	})
 	timeout := time.Until(task.ToTimeout)
 	select {
-	case result := <-resultChan:
-		if result.Err == nil {
-			slr := result.Val.(*stmtctx.StatsLoadResult)
-			if slr.Error != nil {
-				return task, slr.Error
-			}
-			task.ResultCh <- *slr
+	case sr := <-resultChan:
+		// sr.Val is always nil.
+		if sr.Err == nil {
+			task.ResultCh <- result
+			return nil, nil
+		}
+		if !updateNeededItemTaskRetryCountAndCheck(task) {
+			result.Error = sr.Err
+			task.ResultCh <- result
 			return nil, nil
 		}
-		return task, result.Err
+		return task, sr.Err
 	case <-time.After(timeout):
+		if !updateNeededItemTaskRetryCountAndCheck(task) {
+			result.Error = errors.New("stats loading timeout")
+			task.ResultCh <- result
+			return nil, nil
```

Comment:
Do we need to return an error here? Or does this mean we want to ignore the task immediately?

Reply:
want to ignore the task immediately. If the task is essential, it will be inserted into the channel. we should keep a high QPS for HandleOneTask